app/build.gradle.kts

```diff
@@ -183,6 +184,10 @@ dependencies {
 
     // opening hours parser
     implementation("ch.poole:OpeningHoursParser:0.26.0")
+
+    // measuring distance with AR
+    implementation("com.google.ar:core:1.29.0")
+    implementation("com.google.ar.sceneform:core:1.17.1")
```

Comment:
Will StreetComplete still be eligible for F-Droid when this is included? Or will it need another build flavor?

Reply:
For some apps F-Droid maintains patches applied before build.

Reply:
I don't know, we will see what happens. Maybe it will just get a flag "contains non-free software" or something(?)

In any case, shouldn't be hard (for them) to remove this functionality with patches.

Reply:
details><summary>GRR GITHUB</summary>

I had written a whole comment here but not yet posted, then clicked "unresolve conversation", and GitHub decided to delete my comment. But anyway.
</details>

My understanding is that F-Droid does not knowingly distribute non-free apps, period. So, once someone noticed, they would probably remove F-Droid, unless someone cares enough to write a patch.

I care enough to write a patch. Actually, I've previously considered volunteering to be the "SC F-Droid [maintainer](http://kmkeen.com/maintainers-matter/)" if the need should arise. It seems that time may be here. However, I must first raise an objection to adding a dependency on Google Play Services, and especially to including a non-free library in the app. Yes, we're already using Android and thus, trusting Google, but I think proprietary dependencies and shipping proprietary code are both meaningful lines that I'd rather not cross. Of course, I know it isn't my decision to make.

If you decide to continue down this route, I'd rather keep the patches in this repo than in the F-Droid repo. That way, you only need to get the necessary code from one location. I'm not sure if it would be best to maintain a patchset, use a separate `foss`/`fdroid` branch, or create a new build flavor.

Reply:
Well, mainline (`master`) will be the release on github and on google play, signed with my private key.

Maybe it is possible to exclude dependencies via a gradle build config constant. E.g. the `AddCyclewayWidth` quest could exist without AR, just the button to start the AR activity would be missing.

Reply:
Actually, all of [Google Filament](https://github.com/google/filament) (rendering engine), [Google Sceneform](https://github.com/google-ar/sceneform-android-sdk) (connecting ARCore/Filament easily) and the [AR Core SDK](https://github.com/google-ar/arcore-android-sdk) are open source. 

What is not open source is the ARCore (aka "Google Play Services for AR") app. That app does all the heavy lifiting, image processing etc.
The AR Core SDK just communicates with this app and makes sure that that app is installed and up-to-date before doing anything useful. But **communicating** with a closed source app doesn't sound like it would violate against F-Droid guidelines.
If the user cannot install the app because the Google Play Services are not installed at all (such in the case of @FloEdelmann ) or the user doesn't want to install the app, it will simply do nothing. 🤷

Reply:
Ah, cool! I didn't realize— usually it's Android components which are Open Source and everything Google-branded is proprietary, so I didn't look further.

Yes, that should be fine for F-Droid. There might be an antifeature tag (similar to `NonFreeNet` for apps which use non-free network services), but I'm not sure. A 'wait and see' approach should be fine, in that case.